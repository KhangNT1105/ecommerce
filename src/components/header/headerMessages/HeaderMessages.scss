@import '../../../styles/variables';

/* stylelint-disable */
.headerMessages {
  &--search {
    &__append {
      border-left: 1px solid $border-input;
    }

    &__icon {
      display: flex;
      justify-content: center;
      align-items: center;
      padding: 0.375rem 0.75rem;
      border: 1px solid $border-input;
      position: relative;
      border-top-right-radius: 0.25rem;
      border-bottom-right-radius: 0.25rem;
    }
  }
}
