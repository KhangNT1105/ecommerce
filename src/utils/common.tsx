import React from 'react';
import { Denomination } from 'Props.d';
import { assign, clone, map, reduce, isObjectLike } from 'lodash';
import { validateRequired, validateEmail } from 'utils/validation';
import { ACTIVATE, ENTITY_STATUS_FROM_SERVER, DEACTIVATE, DELETE, APPROVE, REJECT } from '../constants';
import { Genre, SubGenre } from 'stores/ConfigurationStore/configurationType.d';

export const checkRenderDoubleValue = (value1?: string | number, value2?: string | number, value3?: any | '') => {
  let minus = '';
  if (value1 && value2) {
    minus = ' - ';
  }
  return (
    <>
      {value1} {value3 && <b>{value3}</b>}
      {minus}
      {value2}
    </>
  );
};

export type SelectOptionProps = {
  id: string;
  value: string;
  label: string;
};

type DataProps = {
  [key: string]: any;
};

export type SelectOptionKeyMappingProps = {
  key: string;
  value: string;
  label: string;
};

export const renderSelectOptions = (data: DataProps[], keyMapping?: SelectOptionKeyMappingProps) => {
  return (
    <>
      <option value="">Select</option>
      {data.map((item) => {
        if (keyMapping && Object.entries(keyMapping).length > 0) {
          return (
            <option key={item[keyMapping.key]} value={item[keyMapping.value]}>
              {item[keyMapping.label]}
            </option>
          );
        }
        return (
          <option key={item.id} value={item.value}>
            {item.label}
          </option>
        );
      })}
    </>
  );
};

export const checkRenderArray = (items: string[]) => {
  return <span>{items.join(', ')}</span>;
};

export const getListPrices = (denominations: Denomination[]) => {
  const listPrices = denominations.map((item) => item.price);
  return listPrices.sort((a, b) => a - b);
};
export const getPrices = (productDetail: any) => {
  if (productDetail.valueType === 'RANGE') {
    return checkRenderDoubleValue(productDetail.minPrice, productDetail.maxPrice);
  }
  if (productDetail.valueType === 'SLAB') {
    const values = getListPrices(productDetail.denominations);
    return values.join(', ');
  }
};

export interface IPagingModelOption {
  pageIndex: number;
  pageSize: number;
}

export interface IPagingModelResult<T> {
  items: T[];
  hasPrevPage: boolean;
  hasNextPage: boolean;
  prevPage: number | null;
  nextPage: number | null;
  size: number;
  page: number;
  totalItems: number;
  totalPages: number;
}

export function getPaginationResponse<T>(
  pagingModel: IPagingModelOption,
  result: T[],
  totalItems: number
): IPagingModelResult<T> {
  const totalPages = Math.floor((totalItems + pagingModel.pageSize - 1) / pagingModel.pageSize);
  const pagingModelResult: IPagingModelResult<T> = {
    items: result,
    totalItems,
    totalPages,
    hasPrevPage: pagingModel.pageIndex > 0,
    hasNextPage: pagingModel.pageIndex < totalPages - 1,
    prevPage: pagingModel.pageIndex > 0 ? pagingModel.pageIndex - 1 : null,
    nextPage: pagingModel.pageIndex < totalPages - 1 ? pagingModel.pageIndex + 1 : null,
    size: pagingModel.pageSize,
    page: pagingModel.pageIndex
  };
  return pagingModelResult;
}

export function replaceSpecialCharacters(str: string, character: string) {
  return str.replace(/(?:\.(?![^.]+$)|[^\w.])+/g, character);
}

export const classConcat = (classes: string[]): string => {
  if (classes.length) {
    return classes.filter((className) => !!className).join(' ');
  }
  return '';
};

export const recursiveGetFieldName = (obj: object, entityField: string, fieldNameCombine?: string): any => {
  let listFieldConfig: any;
  listFieldConfig = reduce(
    obj,
    (result, value, key) => {
      let cloneResult = clone(result);
      let localFieldName: any = clone(fieldNameCombine);
      if (key === entityField) {
        map(value, (item: any, k) => {
          const getFieldName = `${localFieldName}.${key}.${k}`;

          // check whether any validation
          if (item.validationRules && item.validationRules.length > 0) {
            item.validationRules.map((itemValidate: any, index: number) => {
              if (itemValidate.rule === 'isRequired') {
                item.validationRules[index].validator = validateRequired;
              }
              if (itemValidate.rule === 'isEmail') {
                item.validationRules[index].validator = validateEmail;
              }

              return itemValidate;
            });
          }

          // insert key, value into list field
          return (cloneResult = {
            ...cloneResult,
            [getFieldName]: item
          });
        }); // reset field name

        localFieldName = '';
      } else if (isObjectLike(value)) {
        if (!localFieldName) {
          localFieldName = key;
        } else {
          localFieldName += `.${key}`;
        } // recall recursive fn if not be found the expected entityField

        return assign(cloneResult, recursiveGetFieldName(value, entityField, localFieldName));
      }

      listFieldConfig = {
        ...listFieldConfig,
        ...cloneResult
      };

      return listFieldConfig;
    },
    {}
  );

  return listFieldConfig;
};

export type ObjectFieldType = {
  [field: string]: string | object | string[] | object[] | any;
};

export const trimAllValueWithObjectField = (objField: ObjectFieldType) => {
  if ((!Array.isArray(objField) && typeof objField !== 'object') || objField === null) {
    return objField;
  }
  return Object.keys(objField).reduce(
    (acc: ObjectFieldType, key) => {
      acc[key] =
        typeof objField[key] === 'string'
          ? (objField[key] as string).trim()
          : trimAllValueWithObjectField(objField[key]);
      return acc;
    },
    Array.isArray(objField) ? [] : {}
  );
};

const listShowBtnActivate = [
  ENTITY_STATUS_FROM_SERVER.INACTIVATE,
  ENTITY_STATUS_FROM_SERVER.DRAFT,
  ENTITY_STATUS_FROM_SERVER.UPDATED,
  ENTITY_STATUS_FROM_SERVER.ACTIVATE
];

const listShowBtnSMActivate = [
  ENTITY_STATUS_FROM_SERVER.INACTIVATE,
  ENTITY_STATUS_FROM_SERVER.DRAFT,
  ENTITY_STATUS_FROM_SERVER.UPDATED
];

const listShowBtnSMDelete = [ENTITY_STATUS_FROM_SERVER.DRAFT, ENTITY_STATUS_FROM_SERVER.INACTIVATE];

const listShowBtnDeactivate = [ENTITY_STATUS_FROM_SERVER.ACTIVATE, ENTITY_STATUS_FROM_SERVER.UPDATED];

const listShowBtnDelete = [ENTITY_STATUS_FROM_SERVER.INACTIVATE, ENTITY_STATUS_FROM_SERVER.DRAFT];

export const checkEntityActionNonApproval = (action: string, entityStatus: string) => {
  switch (action) {
    case ACTIVATE:
      return listShowBtnActivate.includes(entityStatus);
    case DEACTIVATE:
      return listShowBtnDeactivate.includes(entityStatus);
    case DELETE:
      return listShowBtnDelete.includes(entityStatus);
    default:
      return true;
  }
};

export const checkSMEntityActionNonApproval = (action: string, entityStatus: string) => {
  switch (action) {
    case ACTIVATE:
      return listShowBtnSMActivate.includes(entityStatus);
    case DEACTIVATE:
      return listShowBtnDeactivate.includes(entityStatus);
    case DELETE:
      return listShowBtnSMDelete.includes(entityStatus);
    default:
      return true;
  }
};

export const getAllSubGenres = (genres: Genre[]) =>
  genres
    .map(({ children }) => children)
    .reduce((currentValue, subGenres) => {
      if (subGenres) {
        return [...currentValue, ...subGenres];
      }
      return currentValue;
    }, []);

export const getAllMicroGenres = (subGenres: SubGenre[]) =>
  subGenres
    .map(({ children }) => children)
    .reduce((currentValue, microGenre) => {
      if (microGenre) {
        return [...currentValue, ...microGenre];
      }
      return currentValue;
    }, []);

const listShowBtnActivateWithApproval = [...listShowBtnActivate, ENTITY_STATUS_FROM_SERVER.REJECT];
const listShowBtnDeactivateWithApproval = [...listShowBtnDeactivate];
const listShowBtnDeleteWithApproval = [...listShowBtnDelete];
const listShowBtnApproveWithApproval = [ENTITY_STATUS_FROM_SERVER.APPROVAL_PENDING];
const listShowBtnRejectWithApproval = [ENTITY_STATUS_FROM_SERVER.APPROVAL_PENDING];

export const checkEntityActionApproval = (action: string, entityStatus: string) => {
  switch (action) {
    case ACTIVATE:
      return listShowBtnActivateWithApproval.includes(entityStatus);
    case DEACTIVATE:
      return listShowBtnDeactivateWithApproval.includes(entityStatus);
    case DELETE:
      return listShowBtnDeleteWithApproval.includes(entityStatus);
    case APPROVE:
      return listShowBtnApproveWithApproval.includes(entityStatus);
    case REJECT:
      return listShowBtnRejectWithApproval.includes(entityStatus);
    default:
      return true;
  }
};

export function createMarkup() {
  return {
    __html: `<div class="zalo-chat-widget" data-oaid="927003371386666481" data-welcome-message="Rất vui khi được hỗ trợ bạn!" data-autopopup="0" data-width="450" data-height="450"></div>
    `
  };
}
