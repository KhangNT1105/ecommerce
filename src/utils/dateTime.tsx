import {
  format,
  parse,
  getTime,
  subDays,
  subMonths,
  startOfWeek,
  endOfWeek,
  startOfMonth,
  endOfMonth,
  startOfToday,
  endOfToday,
  startOfDay,
  endOfDay
} from 'date-fns';

import { DATE_FORMAT, DATE_TIME_FULL_FORMAT } from 'constants/index';

export type DateRangeType = { from: string | null; to: string | null };

export const formatDate = (timestamp: string) => {
  return format(parse(timestamp, 'T', new Date()), DATE_FORMAT);
};

export type DateRangeProps = {
  dateFrom: string | null;
  dateTo: string | null;
};

/**
 * get timestamp from a string
 * @param: value string
 */
export const getTimestamp = (value: string): number => {
  return getTime(parse(value, DATE_FORMAT, new Date()));
};

export const getUTCDateFn = (date: Date) => {
  return new Date(
    date.getUTCFullYear(),
    date.getUTCMonth(),
    date.getUTCDate(),
    date.getUTCHours(),
    date.getUTCMinutes(),
    date.getUTCSeconds()
  );
};

/**
 * Description: combine day from DatePicker and local time then convert to UTC Date
 * @param date: from date picker
 * @return: UTC Date
 */
export const convertDatePickerToUTC = (date: Date) => {
  const today = new Date();
  const combineDateTime = new Date(
    date.getFullYear(),
    date.getMonth(),
    date.getDate(),
    today.getHours(),
    today.getMinutes(),
    today.getSeconds()
  );

  return getUTCDateFn(combineDateTime);
};

/**
 * Description: convert HH:MM:SS string to seconds
 * @param dateString
 * @param second
 */
export const convertTimeStringToMillisecond = (dateString: String) => {
  return (dateString as any).split(':').reduce((acc: any, time: any) => 60 * acc + +time) * 1000;
};

export const convertDateFromToUTC = (date: Date) => {
  const combineDateTime = new Date(date.getFullYear(), date.getMonth(), date.getDate());

  return getUTCDateFn(startOfDay(combineDateTime));
};

export const convertDateToToUTC = (date: Date) => {
  const combineDateTime = new Date(date.getFullYear(), date.getMonth(), date.getDate());

  return getUTCDateFn(endOfDay(combineDateTime));
};

export const getToday = (): DateRangeProps => {
  const dateFrom = getUTCDateFn(startOfToday());
  const dateTo = getUTCDateFn(endOfToday());

  return {
    dateFrom: format(dateFrom, DATE_TIME_FULL_FORMAT),
    dateTo: format(dateTo, DATE_TIME_FULL_FORMAT)
  };
};

export const getLast7days = (): DateRangeProps => {
  const dateTo = format(getUTCDateFn(endOfToday()), DATE_TIME_FULL_FORMAT);
  const dateFrom = format(getUTCDateFn(subDays(startOfToday(), 7)), DATE_TIME_FULL_FORMAT);

  return { dateFrom, dateTo };
};

export const getLast30days = (): DateRangeProps => {
  const dateTo = format(getUTCDateFn(endOfToday()), DATE_TIME_FULL_FORMAT);
  const dateFrom = format(getUTCDateFn(subDays(startOfToday(), 30)), DATE_TIME_FULL_FORMAT);

  return { dateFrom, dateTo };
};

export const getLastWeek = (): DateRangeProps => {
  const dateFrom = format(getUTCDateFn(startOfWeek(subDays(startOfToday(), 7))), DATE_TIME_FULL_FORMAT);
  const dateTo = format(getUTCDateFn(endOfWeek(subDays(endOfToday(), 7))), DATE_TIME_FULL_FORMAT);

  return { dateFrom, dateTo };
};

export const getLastMonth = (): DateRangeProps => {
  const dateFrom = format(getUTCDateFn(startOfMonth(subMonths(startOfToday(), 1))), DATE_TIME_FULL_FORMAT);
  const dateTo = format(getUTCDateFn(endOfMonth(subMonths(endOfToday(), 1))), DATE_TIME_FULL_FORMAT);

  return { dateFrom, dateTo };
};

export const convertMillisecondToHHMMSS = (milliseconds: number) => {
  if (!milliseconds) return '';
  const date = new Date(milliseconds);
  const hh = date.getUTCHours();
  const mm = date.getUTCMinutes();
  const ss = date.getSeconds();
  const hours = hh < 10 ? `0${hh}` : `${hh}`;
  const minutes = mm < 10 ? `0${mm}` : `${mm}`;
  const seconds = ss < 10 ? `0${ss}` : `${ss}`;

  return `${hours}:${minutes}:${seconds}`;
};
