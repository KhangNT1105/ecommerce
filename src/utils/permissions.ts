import { matchPath } from 'react-router';
import { FieldType, PermissionFieldsType, FormPermissionsType } from '../components/roleFormField/RoleFormFields.d';
import { BUTTONS_VISIBILITY, PERMISSION_PREFIX, PAGE_ACCESSIBILITY } from '../constants/enum';
/**
 * Description: check user permissions have spec permissions
 * @param permissions: array of user permissions
 * @param modes: array of PERMISSION_PREFIX
 * Results: boolean
 */
export const hasModes = (permissions: string[], modes: string[]) => {
  const foundMode = permissions.filter((item) => {
    const includes = modes.filter((mode) => {
      return item === mode;
    });
    return includes.length > 0;
  });
  return foundMode.length === modes.length;
};

/**
 * Description: check user permissions only have 1 permission
 * @param permissions: array of user permissions
 * @param mode: PERMISSION_PREFIX
 * Results: boolean
 */
export const hasOnlyMode = (permissions: string[], mode: string) => {
  const viewMode = permissions.filter((item) => {
    return item === mode;
  });
  return permissions.length === 1 && viewMode.length === 1;
};

/**
 * Description: generate new structure data:
 * Params:
 *    @param formPermissions: array of permissions and element name (id)
 *    @param userPermissions: array of permissions of logged in users
 *
 * Results:
 * {
 *  btnSave: 'show',
 *  btnCancel: 'NA',
 *  ...
 * }
 */
export const reGeneratePermissionFields = (formPermissions: FormPermissionsType[], userPermissions: string[]) => {
  const permissionFields: PermissionFieldsType = formPermissions
    .filter(({ permissions }) => {
      return userPermissions.includes(permissions);
    })
    .map((formPermissions: FormPermissionsType) => {
      const { permissions, ...restPermissions } = formPermissions;
      return restPermissions;
    });

  const fieldsWithPermissions = permissionFields.reduce(
    (accumulator: FieldType, currentValue: FieldType, index: number) => {
      const fields = {
        ...accumulator
      };
      Object.keys(currentValue).map((operation: string) => {
        if (currentValue[operation] === BUTTONS_VISIBILITY.SHOW) {
          fields[operation] = currentValue[operation];
        }
        return true;
      });
      return {
        ...fields
      };
    },
    {}
  );
  return fieldsWithPermissions;
};
/**
 * Description: get accessibility of page with specification page
 * Params:
 *    @param formPermissions: array of permissions of page
 *    @param userPermissions: array of permissions of logged in users
 *    @param strPermission: permission
 *
 * Results:
 * pageAccessibility: boolean
 */
export const checkPageAccessibility = (
  formPermissions: FormPermissionsType[],
  userPermissions: string[],
  strPermission: string
) => {
  const pagePermissions = reGeneratePermissionFields(formPermissions, userPermissions);
  return pagePermissions[strPermission] === PAGE_ACCESSIBILITY.SHOW;
};

/**
 * Description: get accessibility of page with specification entity
 * Params:
 *    @param path: current path of site
 *    @param entityName: entity
 *    @param formPermissionsCreate: array of permissions of create page
 *    @param formPermissionsEdit: array of permissions of edit page
 *    @param formPermissionsList: array of permissions of list page
 *    @param pathCreate: route string of create page
 *    @param pathEdit: route string of edit page
 *    @param userPermissions: array of permissions of logged in users
 *
 * Results:
 * pageAccessibility: boolean
 */
export const getPageAccessibility = (
  path: string,
  entityName: string,
  formPermissionsCreate: FormPermissionsType[],
  formPermissionsEdit: FormPermissionsType[],
  formPermissionsList: FormPermissionsType[],
  pathCreate: string,
  pathEdit: string,
  userPermissions: string[]
) => {
  let pageAccessibility = true;
  if (matchPath(path, { path: pathCreate, exact: true })) {
    pageAccessibility = checkPageAccessibility(
      formPermissionsCreate,
      userPermissions,
      `${entityName}_${PERMISSION_PREFIX.CREATE.toLowerCase()}`
    );
  } else if (matchPath(path, { path: pathEdit, exact: true })) {
    pageAccessibility = checkPageAccessibility(
      formPermissionsEdit,
      userPermissions,
      `${entityName}_${PERMISSION_PREFIX.EDIT.toLowerCase()}`
    );
  } else {
    pageAccessibility = checkPageAccessibility(formPermissionsList, userPermissions, `${entityName}`);
  }
  return pageAccessibility;
};

/**
 * Description: check page is accessible by entity
 * Params:
 *    @param page: page name
 *    @param path: current path of site
 *    @param userPermissions: array of permissions of logged in users
 *
 * Results:
 * pageAccessibility: boolean
 */
export const checkPagePermissisons = (page: string, path: string, userPermissions: string[]) => {
  // let pagePermissions: { [permission: string]: string };
  const pageAccessibility = true;

  return pageAccessibility;
};

/**
 * Description: get form permissions of list page by entity
 * Params:
 *    @param entity: entity name
 *
 * Results:
 * formPermissions: FormPermissionsType[]
 */
export const getFormPermissionsPageList = (entity: string) => {};
