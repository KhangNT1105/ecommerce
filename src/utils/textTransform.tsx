export const upperCaseFirst = (str: String = '') => {
  return str.toLowerCase().replace(/^\w/, (chars) => chars.toUpperCase());
};
