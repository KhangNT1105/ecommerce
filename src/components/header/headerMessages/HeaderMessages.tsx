import React from 'react';
import {
  UncontrolledDropdown,
  DropdownToggle,
  InputGroup,
  Input,
  InputGroupAddon,
  ListGroupItem,
  ListGroup,
  Media
} from 'reactstrap';

import IconWithBadge from '../../iconWithBadge/IconWithBadge';
import { Dropdown, DropdownSection } from '../../dropdown';
import './HeaderMessages.scss';
import { HeaderMessagesProps } from './HeaderMessages.d';

const messagesColors = {
  success: 'text-success',
  danger: 'text-danger',
  warning: 'text-warning'
};

const HeaderMessages: React.FC<HeaderMessagesProps> = ({ messages }) => {
  return (
    <UncontrolledDropdown nav={true} inNavbar={true}>
      <DropdownToggle nav={true}>
        <IconWithBadge badgeNumber={messages.length} badgeBackground="secondary">
          <i className="fa fa-envelope-o fa-fw" />
        </IconWithBadge>
      </DropdownToggle>
      <Dropdown right={true}>
        <DropdownSection className="d-flex justify-content-between align-items-center">
          <h6 className="mb-0">Messages</h6>
          <i className="fa fa-pencil" />
        </DropdownSection>
        <DropdownSection>
          <InputGroup>
            <Input placeholder="Search Messages..." className="headerMessages--search__input" />
            <InputGroupAddon className="headerMessages--search__append" addonType="append">
              <div className="headerMessages--search__icon">
                <i className="fa fa-search" />
              </div>
            </InputGroupAddon>
          </InputGroup>
        </DropdownSection>

        <DropdownSection list={true}>
          <ListGroup>
            {messages.map((message, index) => (
              <ListGroupItem key={`${message.time}-${index}`} action={true}>
                <Media>
                  <Media left={true}>
                    <div className="avatar-image avatar-image--loaded mr-4">
                      <div className="avatar avatar--md avatar-image__image">
                        <div className="avatar__content">
                          <img src={message.avatar} alt="" />
                        </div>
                      </div>
                    </div>
                  </Media>
                  <Media body={true}>
                    <span className="d-flex justify-content-start">
                      <i
                        className={`fa fa-circle small ${
                          messagesColors[message.status as keyof typeof messagesColors]
                        } mr-2 d-flex align-items-center`}
                      />
                      <span className="h6 pb-0 mb-0 d-flex align-items-center">{message.username}</span>

                      <span className="ml-1 small">(23)</span>
                      <span className="ml-auto small">{message.time}</span>
                    </span>
                    <p className="mt-2 mb-1">{message.content}</p>
                  </Media>
                </Media>
              </ListGroupItem>
            ))}
          </ListGroup>
        </DropdownSection>

        <DropdownSection className="text-center">
          View All
          <i className="fa fa-angle-right fa-fw ml-2" />
        </DropdownSection>
      </Dropdown>
    </UncontrolledDropdown>
  );
};

export default HeaderMessages;
