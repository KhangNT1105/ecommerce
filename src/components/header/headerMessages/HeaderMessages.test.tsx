import { testSnapshots } from '../../../utils/test';
import HeaderMessages from './HeaderMessages';

describe('<HeaderMessages />', () => {
  testSnapshots(HeaderMessages, [
    {
      props: {
        messages: [
          {
            avatar: 'avatar',
            username: 'John',
            status: 'Active',
            time: '23/04/2020',
            content: 'content'
          }
        ]
      },
      description: 'render snapshot with messages'
    }
  ]);
});
