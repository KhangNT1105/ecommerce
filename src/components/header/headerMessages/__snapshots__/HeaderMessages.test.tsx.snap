// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderMessages /> snapshots render snapshot with messages 1`] = `
<li
  className="dropdown nav-item"
  onKeyDown={[Function]}
>
  <a
    aria-expanded={false}
    aria-haspopup={true}
    className="nav-link"
    href="#"
    onClick={[Function]}
  >
    <div
      className="icon-with-badge undefined"
    >
      <i
        className="fa fa-envelope-o fa-fw"
      />
      <span
        className="icon-with-badge__badge badge badge-secondary badge-pill"
      >
        1
      </span>
    </div>
  </a>
  <div
    aria-hidden={true}
    className="extended-dropdown dropdown-menu dropdown-menu-right"
    role="menu"
    style={
      Object {
        "width": "100vw",
      }
    }
    tabIndex="-1"
  >
    <div
      className="extended-dropdown__section d-flex justify-content-between align-items-center"
    >
      <h6
        className="mb-0"
      >
        Messages
      </h6>
      <i
        className="fa fa-pencil"
      />
    </div>
    <div
      className="extended-dropdown__section"
    >
      <div
        className="input-group"
      >
        <input
          className="headerMessages--search__input form-control"
          placeholder="Search Messages..."
          type="text"
        />
        <div
          className="headerMessages--search__append input-group-append"
        >
          <div
            className="headerMessages--search__icon"
          >
            <i
              className="fa fa-search"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="extended-dropdown__section extended-dropdown__section--list"
    >
      <ul
        className="list-group"
      >
        <li
          className="list-group-item-action list-group-item"
        >
          <div
            className="media"
          >
            <div
              className="media-left"
            >
              <div
                className="avatar-image avatar-image--loaded mr-4"
              >
                <div
                  className="avatar avatar--md avatar-image__image"
                >
                  <div
                    className="avatar__content"
                  >
                    <img
                      alt=""
                      src="avatar"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="media-body"
            >
              <span
                className="d-flex justify-content-start"
              >
                <i
                  className="fa fa-circle small undefined mr-2 d-flex align-items-center"
                />
                <span
                  className="h6 pb-0 mb-0 d-flex align-items-center"
                >
                  John
                </span>
                <span
                  className="ml-1 small"
                >
                  (23)
                </span>
                <span
                  className="ml-auto small"
                >
                  23/04/2020
                </span>
              </span>
              <p
                className="mt-2 mb-1"
              >
                content
              </p>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="extended-dropdown__section text-center"
    >
      View All
      <i
        className="fa fa-angle-right fa-fw ml-2"
      />
    </div>
  </div>
</li>
`;
